extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts

```diff
@@ -0,0 +1,601 @@
+			.withProps({ ariaLive: 'polite' })
+			.component();
+		this._rootContainer.addItem(this._assessmentComponent, { flex: '0 0 auto' });
+		this._rootContainer.addItem(this._formContainer.component(), { flex: '0 0 auto' });
+
+		this._disposables.push(this._view.onClosed(
+			e => this._disposables.forEach(
+				d => { try { d.dispose(); } catch { } })));
+
+		await this._view.initializeModel(this._rootContainer);
+	}
+
+	private createStatusComponent(view: azdata.ModelView): azdata.TextComponent {
+		const component = view.modelBuilder.text()
+			.withProps({
+				CSSStyles: {
+
+					'font-size': '13px',
+					'font-weight': '400',
+					'line-height': '18px',
+					'text-decoration': 'none',
+					'margin-left': '8px'
+				}
+			}).component();
+		return component;
+	}
+
+	// Creates the Assessment summary container.
+	private createTargetSummaryComponent(view: azdata.ModelView): FlexContainer {
+		const chooseYourTargetText = this._view.modelBuilder.text().withProps({
+			value: constants.SKU_RECOMMENDATION_CHOOSE_A_TARGET,
+			CSSStyles: {
+				...styles.SECTION_HEADER_CSS,
+				'margin-bottom': '5px'
+			}
+		}).component();
+
+		this._assessmentSummaryCard = view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'display': 'flex',
+				'flex-direction': 'row',
+			}
+		}).component();
+
+		this._dbAssessmentCard = new AssessmentSummaryCard(this, MigrationTargetType.SQLDB, this.migrationStateModel);
+		this._miAssessmentCard = new AssessmentSummaryCard(this, MigrationTargetType.SQLMI, this.migrationStateModel);
+		this._vmAssessmentCard = new AssessmentSummaryCard(this, MigrationTargetType.SQLVM, this.migrationStateModel);
+
+		this._assessmentSummaryCard.addItems([
+			this._dbAssessmentCard.createAssessmentSummaryCard(view),
+			this._miAssessmentCard.createAssessmentSummaryCard(view),
+			this._vmAssessmentCard.createAssessmentSummaryCard(view)]);
+
+		this._assessmentSummaryCardLoader = this._view.modelBuilder.loadingComponent()
+			.withItem(this._assessmentSummaryCard)
+			.component();
+
+		const component = this._view.modelBuilder.flexContainer()
+			.withItems([chooseYourTargetText, this._assessmentSummaryCardLoader])
+			.component();
+
+		return component;
+	}
+
+	private async constructDetails(): Promise<void> {
+		this.wizard.message = {
+			text: '',
+			level: azdata.window.MessageLevel.Error
+		};
+
+		this._serverName = this.migrationStateModel.serverName || (await getSourceConnectionProfile()).serverName;
+
+		if (this.migrationStateModel._runAssessments) {
+			const errors: string[] = [];
+			await this._setAssessmentState(true, false);
+
+			try {
+				await this.migrationStateModel.getDatabaseAssessments([MigrationTargetType.SQLMI, MigrationTargetType.SQLDB]);
+				const assessmentError = this.migrationStateModel._assessmentResults?.assessmentError;
+				if (assessmentError) {
+					errors.push(`message: ${assessmentError.message}${EOL}stack: ${assessmentError.stack}`);
```

Comment:
please create a new function to return a resource string that handles the error inputs as parameters.  same for line 231,

Reply:
done